components/card_footer.js

```diff
@@ -0,0 +1,228 @@
+  padding-left: 9px !important;
+  border-radius: 0px;
+  border-top: 1px solid #f5f5f5 !important;
+  position: relative !important;
+`;
+const cardBottomTitle = css`
+  padding-left: 15px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+const cardBottomFamilyTitle = css`
+  margin-left: 9px;
+  margin-bottom: 25px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+const ExpansionPanelCss = css`
+  margin-bottom: 0px !important;
+  margin-top: 0px !important;
+`;
+const ExpansionPanelOpen = css`
+  background-color: #f5f5f5 !important;
+`;
+const ExpansionPanelClosed = css`
+  background-color: #f1f7fc !important;
+`;
+const collapse = css`
+  padding-top: 25px !important;
+  padding-left: 15px !important;
+  background-color: #f5f5f5 !important;
+`;
+const children = css`
+  width: 100%;
+`;
+const returnIcon = css`
+  -moz-transform: scaleX(-1);
+  -o-transform: scaleX(-1);
+  -webkit-transform: scaleX(-1);
+  transform: scaleX(-1);
+  float: left;
+  filter: FlipH;
+  -ms-filter: fliph;
+  padding-left: 10px;
+`;
+
+export class CardFooter extends Component {
+  state = {
+    open: false
+  };
+
+  benefitTitle = benefit => {
+    return this.props.t("current-language-code") === "en"
+      ? benefit.vacNameEn
+      : benefit.vacNameFr;
+  };
+  childBenefitNames = (benefit, childBenefits, open) => {
+    const length = childBenefits.length;
+    if (open) {
+      return this.props.t("benefits_b.eligible_open_veteran", {
+        x: this.benefitTitle(benefit)
+      });
+    } else {
+      if (length === 1) {
+        return (
+          this.benefitTitle(benefit) +
+          " " +
+          this.props.t("benefits_b.eligible_for_single", {
+            x: this.benefitTitle(childBenefits[0])
+          })
+        );
+      } else {
+        return (
+          this.benefitTitle(benefit) +
+          " " +
+          this.props.t("benefits_b.eligible_for_multi", {
+            x: length
+          })
+        );
+      }
+    }
+  };
+  toggleOpenState = () => {
+    this.setState(previousState => {
+      return { ...previousState, open: !previousState.open };
+    });
+  };
+
+  render() {
+    const { t, benefit, benefits } = this.props;
+
+    let veteranBenefitIds = [];
+    let familyBenefitIds = [];
+
+    this.props.eligibilityPaths.forEach(ep => {
+      if (ep.patronType === "service-person") {
+        veteranBenefitIds = veteranBenefitIds.concat(ep.benefits);
+      }
+      if (ep.patronType === "family") {
+        familyBenefitIds = familyBenefitIds.concat(ep.benefits);
+      }
+    });
+    const childBenefits = benefit.childBenefits
+      ? benefits.filter(ab => benefit.childBenefits.indexOf(ab.id) > -1)
+      : [];
+    const veteranBenefits = childBenefits.filter(
+      ab => veteranBenefitIds.indexOf(ab.id) > -1
+    );
+    const familyBenefits = childBenefits.filter(
+      ab => familyBenefitIds.indexOf(ab.id) > -1
+    );
+
```

Comment:
This is all so very awesome. Only thing I’d suggest is moving this logic for getting family and veteran benefits to functions that we can make a couple simple tests for.